module.go

```diff
@@ -17,49 +19,84 @@ import (
 )
 
 type Module struct {
-	NetworkName gpbft.NetworkName
-	CertStore   *certs.Store
+	Manifest  Manifest
+	CertStore *certs.Store
 
+	id     gpbft.ActorID
 	ds     datastore.Datastore
 	host   host.Host
 	pubsub *pubsub.PubSub
-	verif  gpbft.Verifier
-	sigs   gpbft.Signer
 	ec     ECBackend
 	log    Logger
 
-	// topic is populated after Run is called
-	topic *pubsub.Topic
+	client moduleClient
+}
+
+type moduleClient struct {
+	gpbft.Verifier
+	gpbft.Signer
+	logger         Logger
+	loggerWithSkip Logger
+
+	// Populated after Run is called
+	messageQueue <-chan *gpbft.GMessage
+	topic        *pubsub.Topic
+}
+
+func (mc moduleClient) BroadcastMessage(ctx context.Context, msg []byte) error {
+	return mc.topic.Publish(ctx, msg)
```

Comment:
note: we may need to make this async.

Reply:
Yeah, for sure, but it will change significantly with message builder.

Reply:
Sg. Then I'd just leave a TODO and do it later.